File: test/CodeGen/t4-autogen.cpp
// ----- Autogen kernel by Phaeton -----

void omp_func(
float *a,
float *b,
float *c,
float *d)
{
  d[0] = a[0] + b[0];
  d[0] = a[0] - b[0];
  d[0] = a[0] * b[0];
  d[0] = a[0] / b[0];
  float t0;
  float t1;
  float t2;
  t2 = a[0] + b[0];
  float t3;
  t3 = c[0] - d[0];
  t1 = t2 * t3;
  t0 = t1 + a[0];
  float t4;
  t4 = c[0] * d[0];
  d[0] = t0 + t4;
}

File: test/CodeGen/t4-autogen.py
# ----- Autogen kernel by Phaeton -----
import numpy as np


scalar = np.TensorType('float64', (False,)*0)
a = np.TensorVariable(scalar)
b = np.TensorVariable(scalar)
c = np.TensorVariable(scalar)
d = np.TensorVariable(scalar)
d = a + b
d = a - b
d = a * b
d = a / b
t2 = a + b
t3 = c - d
t1 = t2 * t3
t0 = t1 + a
t4 = c * d
d = t0 + t4
